### ISSUE TITLE
linear policy?

### ISSUE THREAD

**Issue by zafarali** [aravindr93/mjrl#3]
Hi I got here after reading your paper "Towards Generalization and Simplicity in Continuous Control" and I was wondering if the file https://github.com/aravindr93/mjrl/blob/master/mjrl/policies/gaussian_mlp.py was the one used in your paper? The paper claims to use linear policies but it seems this network in an MLP?

**Comment by aravindr93** [comment]
Hi @zafarali thanks for the interest. The paper indeed uses linear policy. For the paper, we used a bare numpy implementation without any autograd (I am planning to make this public as well, but will likely not do any further development in that repo). To get linear policies with this repo, you can simply strip off the hidden layers in the MLP policy. Please let me know if some further clarification is needed.

**Comment by zafarali** [comment]
Thanks for your reply!

So is it safe to say that no gpu's were needed? How long did your experiments take?

I found this surrogate loss in the REINFORCE code, can you elaborate a bit on this? What does "CPI" stand for and will using this surrogate re-create the vanilla REINFORCE algorithm?
https://github.com/aravindr93/mjrl/blob/7705640fed18f9740e2d4bc9ee79bfee4133b044/mjrl/algos/batch_reinforce.py#L37-L44

**Comment by aravindr93** [comment]
Yes, no GPUs are needed for the OpenAI gym continuous control tasks. This is standard practice in other code bases as well (e.g. rallab).

The experiments can take anywhere between 5 minutes to 2 hours depending on complexity of task (from simple swimmer to complex humanoid) on a single good workstation.

CPI stands for Conservative Policy Iteration (paper: http://www.cs.cmu.edu/~./jcl/papers/aoarl/Final.pdf), which was the paper that originally proposed that surrogate. This gradient of the surrogate will be identical to REINFORCE as proved in that paper. The LR will indeed equal 1 when using on-policy samples. This general code structure is useful for other algorithms that use off-policy samples (e.g. PPO, Q-prop).

**Comment by zafarali** [comment]
Thanks, this is super useful!

**Comment by supratikp** [comment]
Looking at linear_nn_comparison.py, together with a linear policy an mlp is used as the value function approximator. Is this what was used for the experiments in the paper as well?
